### ISSUE TITLE
Uncaught Assertion Error

### ISSUE THREAD

**Issue by EricTheMagician** [arunoda/laika#44]
Currently on laika 0.2.2 and nodejs 0.10.15

When one of my assertion fails, the rest of the tests fail as well with the same errror.

```
assert.js:92
  throw new assert.AssertionError({
        ^
AssertionError: false == true
    at ClientConnector.<anonymous> (/Users/username/meteor/website/tests/eventTest.js:42:11)
    at ClientConnector.EventEmitter.emit (events.js:95:17)
    at onCallback (/usr/local/share/npm/lib/node_modules/laika/lib/connectors/client.js:57:15)
    at Socket.<anonymous> (/usr/local/share/npm/lib/node_modules/laika/node_modules/node-phantom/node-phantom.js:199:7)
    at Socket.EventEmitter.emit [as $emit] (events.js:95:17)
    at SocketNamespace.handlePacket (/usr/local/share/npm/lib/node_modules/laika/node_modules/node-phantom/node_modules/socket.io/lib/namespace.js:335:22)
    at Manager.onClientMessage (/usr/local/share/npm/lib/node_modules/laika/node_modules/node-phantom/node_modules/socket.io/lib/manager.js:488:38)
    at WebSocket.Transport.onMessage (/usr/local/share/npm/lib/node_modules/laika/node_modules/node-phantom/node_modules/socket.io/lib/transport.js:387:20)
    at Parser.<anonymous> (/usr/local/share/npm/lib/node_modules/laika/node_modules/node-phantom/node_modules/socket.io/lib/transports/websocket/default.js:36:10)
    at Parser.EventEmitter.emit (events.js:95:17)`
```

My assertion looks like this:

```
        client.on("ready", function(ret){
            assert.ok(!ret.error);
            done();
        });
```

**Comment by arunoda** [comment]
I didn't exactly get the error. Do you mean if one tests faills, all other tests fails too? Is that?

Can you please show me the code to regenerate the error? Then I could easily investigate on this.

**Comment by EricTheMagician** [comment]
Yes, if one test fails. It produces an error.
All other tests don't run and fail with the same error.

This is the test I was running:

```

    test("is possible", function(done,server,client){
        client.eval( function(){

            Accounts.createUser({
                email:"joe@schmoe.com", 
                password:"letmein", 
                profile:{
                    first_name: "joe",
                    last_name: "schmoe"
                }
            });
            var e = {

            };
            Meteor.call("createEvent", e, function(error, result){
                var ret = {error: error, result: result };
                emit("ready", ret);
            })

        });

        client.on("ready", function(ret){
            assert.equal(ret.error, undefined); //this assert fails
            done();
        });

    });
```

and this is the output from running laika

```
 injecting laika...
  loading phantomjs...
  loading initial app pool...


  Creating Events
    ✓ User must be logged in (1054ms)
    1) is possible

  User
    2) needs to have a name
    3) can create account
    4) must have a unique email address


  1 passing (5 seconds)
  4 failing

  1) Creating Events is possible:
     Uncaught AssertionError: {"error":422,"errorType":"Meteor.Error","message":"Please fill in a headline [422]","reason":"Please fill in a headline"} == "undefined"
      at ClientConnector.<anonymous> (/Users/username/meteor/website/tests/eventTest.js:42:11)
      at ClientConnector.EventEmitter.emit (events.js:95:17)
      at onCallback (/usr/local/share/npm/lib/node_modules/laika/lib/connectors/client.js:57:15)
      at Socket.<anonymous> (/usr/local/share/npm/lib/node_modules/laika/node_modules/node-phantom/node-phantom.js:199:7)
      at Socket.EventEmitter.emit [as $emit] (events.js:95:17)
      at SocketNamespace.handlePacket (/usr/local/share/npm/lib/node_modules/laika/node_modules/node-phantom/node_modules/socket.io/lib/namespace.js:335:22)
      at Manager.onClientMessage (/usr/local/share/npm/lib/node_modules/laika/node_modules/node-phantom/node_modules/socket.io/lib/manager.js:488:38)
      at WebSocket.Transport.onMessage (/usr/local/share/npm/lib/node_modules/laika/node_modules/node-phantom/node_modules/socket.io/lib/transport.js:387:20)
      at Parser.<anonymous> (/usr/local/share/npm/lib/node_modules/laika/node_modules/node-phantom/node_modules/socket.io/lib/transports/websocket/default.js:36:10)
      at Parser.EventEmitter.emit (events.js:95:17)

  2) User needs to have a name:
     Uncaught AssertionError: {"error":422,"errorType":"Meteor.Error","message":"Please fill in a headline [422]","reason":"Please fill in a headline"} == "undefined"
      at ClientConnector.<anonymous> (/Users/username/meteor/website/tests/eventTest.js:42:11)
      at ClientConnector.EventEmitter.emit (events.js:95:17)
      at onCallback (/usr/local/share/npm/lib/node_modules/laika/lib/connectors/client.js:57:15)
      at Socket.<anonymous> (/usr/local/share/npm/lib/node_modules/laika/node_modules/node-phantom/node-phantom.js:199:7)
      at Socket.EventEmitter.emit [as $emit] (events.js:95:17)
      at SocketNamespace.handlePacket (/usr/local/share/npm/lib/node_modules/laika/node_modules/node-phantom/node_modules/socket.io/lib/namespace.js:335:22)
      at Manager.onClientMessage (/usr/local/share/npm/lib/node_modules/laika/node_modules/node-phantom/node_modules/socket.io/lib/manager.js:488:38)
      at WebSocket.Transport.onMessage (/usr/local/share/npm/lib/node_modules/laika/node_modules/node-phantom/node_modules/socket.io/lib/transport.js:387:20)
      at Parser.<anonymous> (/usr/local/share/npm/lib/node_modules/laika/node_modules/node-phantom/node_modules/socket.io/lib/transports/websocket/default.js:36:10)
      at Parser.EventEmitter.emit (events.js:95:17)

  3) User can create account:
     Uncaught AssertionError: {"error":422,"errorType":"Meteor.Error","message":"Please fill in a headline [422]","reason":"Please fill in a headline"} == "undefined"
      at ClientConnector.<anonymous> (/Users/username/meteor/website/tests/eventTest.js:42:11)
      at ClientConnector.EventEmitter.emit (events.js:95:17)
      at onCallback (/usr/local/share/npm/lib/node_modules/laika/lib/connectors/client.js:57:15)
      at Socket.<anonymous> (/usr/local/share/npm/lib/node_modules/laika/node_modules/node-phantom/node-phantom.js:199:7)
      at Socket.EventEmitter.emit [as $emit] (events.js:95:17)
      at SocketNamespace.handlePacket (/usr/local/share/npm/lib/node_modules/laika/node_modules/node-phantom/node_modules/socket.io/lib/namespace.js:335:22)
      at Manager.onClientMessage (/usr/local/share/npm/lib/node_modules/laika/node_modules/node-phantom/node_modules/socket.io/lib/manager.js:488:38)
      at WebSocket.Transport.onMessage (/usr/local/share/npm/lib/node_modules/laika/node_modules/node-phantom/node_modules/socket.io/lib/transport.js:387:20)
      at Parser.<anonymous> (/usr/local/share/npm/lib/node_modules/laika/node_modules/node-phantom/node_modules/socket.io/lib/transports/websocket/default.js:36:10)
      at Parser.EventEmitter.emit (events.js:95:17)

  4) User must have a unique email address:
     Uncaught AssertionError: {"error":422,"errorType":"Meteor.Error","message":"Please fill in a headline [422]","reason":"Please fill in a headline"} == "undefined"
      at ClientConnector.<anonymous> (/Users/username/meteor/website/tests/eventTest.js:42:11)
      at ClientConnector.EventEmitter.emit (events.js:95:17)
      at onCallback (/usr/local/share/npm/lib/node_modules/laika/lib/connectors/client.js:57:15)
      at Socket.<anonymous> (/usr/local/share/npm/lib/node_modules/laika/node_modules/node-phantom/node-phantom.js:199:7)
      at Socket.EventEmitter.emit [as $emit] (events.js:95:17)
      at SocketNamespace.handlePacket (/usr/local/share/npm/lib/node_modules/laika/node_modules/node-phantom/node_modules/socket.io/lib/namespace.js:335:22)
      at Manager.onClientMessage (/usr/local/share/npm/lib/node_modules/laika/node_modules/node-phantom/node_modules/socket.io/lib/manager.js:488:38)
      at WebSocket.Transport.onMessage (/usr/local/share/npm/lib/node_modules/laika/node_modules/node-phantom/node_modules/socket.io/lib/transport.js:387:20)
      at Parser.<anonymous> (/usr/local/share/npm/lib/node_modules/laika/node_modules/node-phantom/node_modules/socket.io/lib/transports/websocket/default.js:36:10)
      at Parser.EventEmitter.emit (events.js:95:17)



  cleaning up injected code
```

**Comment by arunoda** [comment]
Thanks. I will work on this.
Today is a busy day! Hope you don't mind working this on tomorrow.

**Comment by EricTheMagician** [comment]
No worries.
I work on my meteor website every few days.

I'm just glad you keep on fixing the bugs that I find :)
(and making meteor better overall)

**Comment by arunoda** [comment]
I'm liable for every piece of code I wrote :) But time is the barrier.

**Comment by arunoda** [comment]
I could re-produce the error, I'm working on a fix.

But there is a quick fix for now. Use `client.once` instead of `client.on`

**Comment by arunoda** [comment]
Fixed and published in version `0.2.24`
